NuRadioReco/examples/PhasedArray/SNR_curves/T02RunSNR_RNOG.py

```diff
@@ -0,0 +1,280 @@
+filt2_highres = channelBandPassFilter.get_filter(fff, 0, 0, None, passband=[80 * units.MHz, 230 * units.MHz], filter_type="cheby1", order=4, rp=.1)
+filt_highres = filt1_highres * filt2_highres
+bandwidth = np.trapz(np.abs(filt_highres) ** 2, fff)
+Vrms_ratio = np.sqrt(bandwidth / (max_freq - min_freq))
+
+if(n_channels == 4):
+    upsampling_factor = 2
+    phasing_angles = np.arcsin(np.linspace(np.sin(main_low_angle), np.sin(main_high_angle), 11))
+    channels = np.arange(4, 8)
+    threshold = 2.63
+    phase = True
+elif(n_channels == 8):
+    upsampling_factor = 4
+    phasing_angles = np.arcsin(np.linspace(np.sin(main_low_angle), np.sin(main_high_angle), 21))
+    channels = None
+    threshold = 2.67
+    phase = True
+elif(n_channels == 1):
+    upsampling_factor = 1
+    phasing_angles = []
+    channels = [4]
+    threshold = 2.02
+    phase = False
+else:
+    print("wrong n_channels!")
+    exit()
+
+def count_events():
+    count_events.events += 1
+count_events.events = 0
+
+class mySimulation(simulation.simulation):
+
+    def _detector_simulation_filter_amp(self, evt, station, det):
+        channelBandPassFilter.run(evt, station, det,
+                                  passband=[0, 500.0 * units.MHz], filter_type='butter', order=10)
+
+    def _detector_simulation_part2(self):
+        # Start detector simulation
+
+        # Convolve efield with antenna pattern
+        efieldToVoltageConverter.run(self._evt, self._station, self._det)
+
+        # Downsample trace back to detector sampling rate
+        new_sampling_rate = 500.0 * units.MHz
+        channelResampler.run(self._evt, self._station, self._det, sampling_rate=new_sampling_rate)
+    
+        # Filter signals
+        channelBandPassFilter.run(self._evt, self._station, self._det, passband=[0.0 * units.MHz, 240.0 * units.MHz],
+                                  filter_type='cheby1', order=9, rp=.1)
+        channelBandPassFilter.run(self._evt, self._station, self._det, passband=[80.0 * units.MHz, 230.0 * units.MHz],
+                                  filter_type='cheby1', order=4, rp=.1)
+
+        # Copying the original traces. Not really needed
+        filtered_signal_traces = {}
+        for channel in self._station.iter_channels():
+            trace = np.array(channel.get_trace())
+            filtered_signal_traces[channel.get_id()] = trace
+
+        # Since there are often two traces within the same thing, gotta be careful
+        Vpps = np.zeros(self._station.get_number_of_channels())
+        for iChan, channel in enumerate(self._station.iter_channels()):
+            trace = np.array(channel.get_trace())
+            Vpps[iChan] = np.max(trace) - np.min(trace)
+
+        factor = 1.0 / (np.mean(Vpps) / 2.0)
+        mult_factors = factor * SNRs
+
+        has_triggered = True
+        while(has_triggered): # search for noise traces that don't set off a trigger
```

Comment:
I don't get what you're doing here. You seem to loop until you found a noise fluctuation that trigger but then you don't do anything with it.

Reply:
It loops until it finds a noise trace that doesn't trigger the phased array, so that noise triggers don't slip into the efficiency plot. It is a bit confusing, it keeps going until has_triggered = false on a noise trace.